spec/presenters/vendor_api/single_application_presenter_spec.rb

```diff
@@ -44,19 +44,33 @@
   end
 
   describe 'attributes.hesa_itt_data' do
-    let(:application_choice) do
-      application_form = create(:completed_application_form, :with_completed_references, :with_equality_and_diversity_data)
-      create(:application_choice, status: 'awaiting_provider_decision', application_form: application_form)
+    context "when an application choice has status 'recruited'" do
+      let(:application_choice) do
+        application_form = create(:completed_application_form, :with_completed_references, :with_equality_and_diversity_data)
+        create(:application_choice, status: 'recruited', application_form: application_form)
+      end
+
+      it 'returns the hesa_itt_data attribute of an application' do
+        equality_and_diversity_data = application_choice.application_form.equality_and_diversity
+        response = VendorAPI::SingleApplicationPresenter.new(application_choice).as_json
+        expect(response.dig(:attributes, :hesa_itt_data)).to eq(
+          disability: equality_and_diversity_data['hesa_disabilities'],
+          ethnicity: equality_and_diversity_data['hesa_ethnicity'],
+          sex: equality_and_diversity_data['hesa_sex'],
+        )
+      end
     end
 
-    it 'returns the hesa_itt_data attribute of an application' do
-      equality_and_diversity_data = application_choice.application_form.equality_and_diversity
-      response = VendorAPI::SingleApplicationPresenter.new(application_choice).as_json
-      expect(response.dig(:attributes, :hesa_itt_data)).to eq(
-        disability: equality_and_diversity_data['hesa_disabilities'],
-        ethnicity: equality_and_diversity_data['hesa_ethnicity'],
-        sex: equality_and_diversity_data['hesa_sex'],
-      )
+    context "when an application choice does not have status 'recruited'" do
+      let(:application_choice) do
+        application_form = create(:completed_application_form, :with_completed_references, :with_equality_and_diversity_data)
+        create(:application_choice, status: 'offer', application_form: application_form)
+      end
+
+      it 'does not return the hesa_itt_data attribute of an application' do
+        response = VendorAPI::SingleApplicationPresenter.new(application_choice).as_json
+        expect(response.dig(:attributes).keys).not_to include(:hesa_itt_data)
```

Comment:
What's the value of a `dig` when we only go one layer?

Reply:
Woops that `dig` should not be I've updates the assertion in this spec. Given we want to assign `nil` to the `:hesa_itt_data` in all other status scenarios it now reads:

` expect(response[:attributes][:hesa_itt_data]).to be_nil`